io_scene_godot/structures.py

```diff
@@ -147,19 +147,53 @@ class NodeTemplate(FileEntry):
     This is a template node that can be used to contruct nodes of any type.
     It is not intended that other classes in the exporter inherit from this,
     but rather that all the exported nodes use this template directly."""
-    def __init__(self, name, node_type, parent_path):
-        if parent_path.startswith("./"):
-            parent_path = parent_path[2:]
-
-        super().__init__(
-            "node",
-            collections.OrderedDict((
-                ("name", name),
-                ("type", node_type),
-                ("parent", parent_path)
-            ))
-        )
-
+    def __init__(self, name, node_type, parent_node):
+        # set child, parent relation
+        self.children = []
+        self.parent = parent_node
+
+        node_name = name.replace('.', '').replace('/', '')
+
+        if parent_node is not None:
+            parent_node.children.append(self)
+
+            parent_path = parent_node.get_path()
+            if parent_path.startswith("./"):
```

Comment:
Should this check be here or in get_path()?

Reply:
yeah, it should be in get_path()